### DISCUSSION TITLE
How can I prevent from unselecting a linked data in a O2M field ?

### DISCUSSION THREAD

**Discussion by ImaCrea** [directus/directus#17915] [Archive]
Hi there,
I'm trying to have field showing a linked data that can't be unselected.

At first I'd use the "read only" value but then when user click on data then she can't modify eventually values inside that linked item, which I'm ok with.

I just don't want user to be able to select another item. So I set off "read only" and I set off "select" and I'm almost there with this : 
![Capture d’écran 2023-03-27 à 18 02 06](https://user-images.githubusercontent.com/764332/227998537-896e05ce-f1cb-42e2-826d-9134a78d4e70.png)

But I'd like user not to be able to delete the item selected. So this same behavior, just without the unselect option would be perfect. Any idea how can I do that please?

Thanks

**Comment by david-zacharias** [comment]
I think it's currently not possible to hide / disable the "x"-Buttons. However, you can restrict users from deleting these relations if you:
- not give permission to _remove_ on the junction table
- set the bahavior for _On Deselct_ to _cascade_ in the relation field settings.

This way configured users can still hit the "x"-Button, but when trying to save the item an error occurs telling they do not have the right to delete the junction item. UX is not nice, but it should work.

<img width="835" alt="Bildschirmfoto 2023-03-27 um 21 29 28" src="https://user-images.githubusercontent.com/1903179/228046979-603f9d8e-1fe1-4838-b0c4-7e9e99098c34.png">

**Reply by ImaCrea** [reply]
Thanks for your suggestion, I'll try that :)
